lib/std/fs/Dir.zig

```diff
@@ -1198,12 +1200,27 @@ fn makeOpenPathAccessMaskW(self: Dir, sub_path: []const u8, access_mask: u32, no
                 component = it.previous() orelse return e;
                 continue;
             },
+            error.PathAlreadyExists => result: {
+                assert(!is_last);
```

Comment:
Am I wrong in thinking this assert is unavoidably trippable? Can't this occur if two threads try to create the last component at the same time?

Reply:
`.create_disposition = if (is_last) w.FILE_OPEN_IF else w.FILE_CREATE,`

As per the documentation, `FILE_CREATE` errors if it already exists but `FILE_OPEN_IF` proceeds to open the existing path.